Makefile

```diff
@@ -1,170 +1,170 @@
+PLUGINS_BASE += XSFeatureNTP.py
+PLUGINS_BASE += XSFeatureQuit.py
+PLUGINS_BASE += XSFeaturePoolEject.py
+PLUGINS_BASE += XSFeaturePoolJoin.py
+PLUGINS_BASE += XSFeaturePoolNewMaster.py
+PLUGINS_BASE += XSFeatureReboot.py
+PLUGINS_BASE += XSFeatureRemoteShell.py
+PLUGINS_BASE += XSFeatureSRCommon.py
+PLUGINS_BASE += XSFeatureSRCreate.py
+PLUGINS_BASE += XSFeatureSRInfo.py
+PLUGINS_BASE += XSFeatureSaveBugReport.py
+PLUGINS_BASE += XSFeatureFullVersion.py
+PLUGINS_BASE += XSFeatureShutdown.py
+PLUGINS_BASE += XSFeatureStatus.py
+PLUGINS_BASE += XSFeatureSuspendSR.py
+PLUGINS_BASE += XSFeatureSyslog.py
+PLUGINS_BASE += XSFeatureSystem.py
+PLUGINS_BASE += XSFeatureTestNetwork.py
+PLUGINS_BASE += XSFeatureTimezone.py
+PLUGINS_BASE += XSFeatureUploadBugReport.py
+PLUGINS_BASE += XSFeatureValidate.py
+PLUGINS_BASE += XSFeatureVMCommon.py
+PLUGINS_BASE += XSFeatureVMInfo.py
+PLUGINS_BASE += XSMenuLayout.py
+
+PLUGINS_OEM :=
+PLUGINS_OEM += XSFeatureClaimSR.py
+PLUGINS_OEM += XSFeatureLicenseNag.py
+PLUGINS_OEM += XSFeatureManagementHelp.py
+PLUGINS_OEM += XSFeatureOEMBackup.py
+PLUGINS_OEM += XSFeatureOEMRestore.py
+PLUGINS_OEM += XSFeatureReset.py
+PLUGINS_OEM += XSFeatureUpdate.py
+PLUGINS_OEM += XSFeatureVerboseBoot.py
+PLUGINS_OEM += XSMenuOEMLayout.py
+
+PLUGINS_EXTRAS :=
+
+ALL_SCRIPTS := $(SCRIPTS)
+ALL_SCRIPTS += $(addprefix plugins-base/, $(PLUGINS_BASE))
+ALL_SCRIPTS += $(addprefix plugins-oem/, $(PLUGINS_OEM))
+ALL_SCRIPTS += $(addprefix plugins-extras/, $(PLUGINS_EXTRAS))
+
+################################################################################
+# Executable:
+COMMAND := xsconsole
+
+################################################################################
+# Documents
+
+#DOCUMENTS :=
+#DOCUMENTS += LICENSE
+
+################################################################################
+install-base:
+	mkdir -p $(LIBDIR)/xsconsole/
+	mkdir -p $(LIBDIR)/xsconsole/plugins-base
+	mkdir -p $(LIBDIR)/xsconsole/plugins-extras
+
+	$(foreach script,$(SCRIPTS),\
+          $(INSTALL) -m $(LIB_MODE) $(script) $(LIBDIR)/xsconsole;)
+
+	$(foreach script,$(PLUGINS_BASE),\
+          $(INSTALL) -m $(LIB_MODE) plugins-base/$(script) $(LIBDIR)/xsconsole/plugins-base;)
+
+	$(foreach script,$(PLUGINS_EXTRAS),\
+          $(INSTALL) -m $(LIB_MODE) plugins-extras/$(script) $(LIBDIR)/xsconsole/plugins-extras;)
+
+	$(INSTALL) -m $(BIN_MODE) $(COMMAND) $(PREFIX)/bin
+
+#	$(foreach docfile,$(DOCUMENTS),\
+#          $(INSTALL) -m $(DOC_MODE) $(docfile) $(DOCDIR);)
+
+	mkdir -p $(PREFIX)/lib/systemd/system
+	$(INSTALL) -m $(LIB_MODE) xsconsole.service $(PREFIX)/lib/systemd/system
+
+install-oem:
+	mkdir -p $(LIBDIR)/xsconsole/plugins-oem
+
+	$(foreach script,$(PLUGINS_OEM),\
+          $(INSTALL) -m $(LIB_MODE) plugins-oem/$(script) $(LIBDIR)/xsconsole/plugins-oem;)
+
+.PHONY: clean
+clean:
+	rm -f *.pyc
+	rm -f */*.pyc
+	rm -rf __pycache__
+
+depend:
+
+.PHONY: test
+test:
+	python -m unittest discover

```

Comment:
Please update the `make test` target to:
```
.PHONY: test
test:
	$(PYTHON) -m unittest discover
```

With this at the top of the file:
```
ifneq (, $(shell which python3))
    PYTHON := python3
else
    PYTHON := python
endif
```

Then it should work on linux systems where `python3` is about or Windows where it is typically installed as `python`.

Reply:
```make
# Best effort attempt to find a python interpreter, defaulting to Python 3 if
# available.  Fall back to just `python` if `which` is nowhere to be found.
PYTHON := $(word 1,$(shell which python3 python python2 2>/dev/null) python)
```